src/uu/install/src/install.rs

```diff
@@ -408,98 +488,85 @@ fn standard(mut paths: Vec<String>, b: Behavior) -> i32 {
         if let Some(parent) = target.parent() {
             if !parent.exists() && b.create_leading {
                 if let Err(e) = fs::create_dir_all(parent) {
-                    show_error!("failed to create {}: {}", parent.display(), e);
-                    return 1;
+                    return Err(InstallError::CreateDirFailed(parent.to_path_buf(), e).into());
                 }
 
                 if mode::chmod(parent, b.mode()).is_err() {
-                    show_error!("failed to chmod {}", parent.display());
-                    return 1;
+                    return Err(InstallError::ChmodFailed(parent.to_path_buf()).into());
                 }
             }
         }
 
         if target.is_file() || is_new_file_path(&target) {
             copy_file_to_file(&sources[0], &target, &b)
         } else {
-            show_error!(
-                "invalid target {}: No such file or directory",
-                target.display()
-            );
-            1
+            Err(InstallError::InvalidTarget(target).into())
         }
     }
 }
 
 /// Copy some files into a directory.
 ///
 /// Prints verbose information and error messages.
-/// Returns an integer intended as a program return code.
+/// Returns a Result type with the Err variant containing the error message.
 ///
 /// # Parameters
 ///
 /// _files_ must all exist as non-directories.
 /// _target_dir_ must be a directory.
 ///
-fn copy_files_into_dir(files: &[PathBuf], target_dir: &Path, b: &Behavior) -> i32 {
+fn copy_files_into_dir(files: &[PathBuf], target_dir: &Path, b: &Behavior) -> UResult<()> {
     if !target_dir.is_dir() {
-        show_error!("target '{}' is not a directory", target_dir.display());
-        return 1;
+        return Err(InstallError::TargetDirIsntDir(target_dir.to_path_buf()).into());
     }
-
-    let mut all_successful = true;
     for sourcepath in files.iter() {
         if !sourcepath.exists() {
-            show_error!(
-                "cannot stat '{}': No such file or directory",
-                sourcepath.display()
+            let err = UIoError::new(
+                std::io::ErrorKind::NotFound,
+                format!("cannot stat '{}'", sourcepath.display()),
             );
-
-            all_successful = false;
+            show!(err);
             continue;
         }
 
         if sourcepath.is_dir() {
-            show_error!("omitting directory '{}'", sourcepath.display());
-            all_successful = false;
+            let err = InstallError::OmittingDirectory(sourcepath.to_path_buf());
+            show!(err);
             continue;
         }
 
         let mut targetpath = target_dir.to_path_buf();
         let filename = sourcepath.components().last().unwrap();
         targetpath.push(filename);
 
-        if copy(sourcepath, &targetpath, b).is_err() {
-            all_successful = false;
+        if let Err(e) = copy(sourcepath, &targetpath, b) {
+            show!(e);
```

Comment:
Not necessary, but there is a `show_if_err!` macro for this purpose. If you think this is clearer, feel free to leave it in

Reply:
You're absolutely right! It's much nicer to read that way, too. Added it.